com.unity.perception/Documentation~/FAQ/FAQ.md

```diff
@@ -0,0 +1,584 @@
+        m_SelectedPrefabs.Clear();
+
+        //Select a random number of prefabs for this frame. Placement calculations will be done based on this subset. 
+        for (var i = 0; i < m_SelectionPoolSizePerFrame; i++)
+        {
+            var randIndex = (int)Mathf.Round((m_IndexSelector.Sample() * prefabs.categories.Count) - 0.5f);
+            m_SelectedPrefabs.Add(prefabs.categories[randIndex].Item1);
+        }
+
+        //Calculate the minimum separation distance needed for the selected prefabs to not overlap.  
+        var separationDistance = CalculateMaxSeparationDistance(m_SelectedPrefabs);
+        var seed = SamplerState.NextRandomState();
+        var placementSamples = PoissonDiskSampling.GenerateSamples(
+            placementArea.x, placementArea.y, separationDistance, seed);
+        var offset = new Vector3(placementArea.x, placementArea.y, 0f) * -0.5f;
+        foreach (var sample in placementSamples)
+        {
+            //Pick a random prefab from the selected subset and instantiate it.
+            var randIndex = (int)Mathf.Round((m_IndexSelector.Sample() * m_SelectedPrefabs.Count) - 0.5f);
+            var instance = m_GameObjectOneWayCache.GetOrInstantiate(m_SelectedPrefabs[randIndex]);
+            instance.transform.position = new Vector3(sample.x, sample.y, depth) + offset;
+            instance.transform.localScale = Vector3.one * m_ScaleFactor;
+        }
+
+        placementSamples.Dispose();
+    }
+
+    protected override void OnIterationEnd()
+    {
+        m_GameObjectOneWayCache.ResetAllObjects();
+    }
+
+    /// <summary>
+    /// Calculates the separation distance needed between placed objects to be sure that no two objects will overlap
+    /// </summary><br>
+    /// <returns>The max separation distance</returns>
+    float CalculateMaxSeparationDistance(ICollection<GameObject> categories)
+    {
+        var maxBoundsSize = m_GameObjectBoundsSizeCache.Where(item => categories.Contains(item.Key)).Max(pair => pair.Value); 
+        return maxBoundsSize * m_ScaleFactor * nonOverlapGuarantee;
+    }
+
+    float CalculateAverageBoundsSize()
+    {
+        foreach (var category in prefabs.categories)
+        {
+            var prefab = category.Item1;
+
+            prefab.transform.localScale = Vector3.one;
+            var renderers = prefab.GetComponentsInChildren<Renderer>();
+            var totalBounds = new Bounds();
+            foreach (var renderer in renderers)
+            {
+                totalBounds.Encapsulate(renderer.bounds);
+            }
+
+            var boundsSize = totalBounds.size.magnitude;
+            m_GameObjectBoundsSizeCache.Add(prefab, boundsSize);
+        }
+
+        return m_GameObjectBoundsSizeCache.Values.Average();
+    }
+}
+
+  ```
+---
+</details>
+
+<details>
+  <summary><strong>Q: What if I don't want randomized object placement? Can I move my objects in a non-randomized deterministic manner on each frame?</strong> </summary><br>
+
```

Comment:
An alternative way to do this is to use a ConstantSampler

Reply:
👍